### PR TITLE
chore(RHTAPWATCH-151): build PipelineRun success and failure events

### PR DESCRIPTION
**Pull request by ShakedAviv50313**
Produce a segment event when a build PipelineRun succeeds or fails.

### CODE REVIEW

**Comment by ShakedAviv50313** [comment]
This is just a draft, I will modify it tomorrow, no need to review this yet

File: querygen/querygen.go
Comment (ifireball) [inline querygen/querygen.go:74]: Looks like you could build the "eval" string using the "reason" field you use in the `where` clause below, and therefore unit these two functions into one.

**Review by ifireball — CHANGES_REQUESTED** [review]

File: querygen/querygen.go
Comment (ifireball) [inline querygen/querygen.go:76]: Since this is very similar to the existing query in GenBuildPipelineRunStartedQuery, could both be combined?

The trouble with `spath` is that you can't see the fields that are being generated in the query string - so things are less readable. Perhaps we could give up on the first `spath` and run `mvexpand` directly on the nested field?

As for the 2nd `spath` could we do without it? As far as I can tell, it would make the "where" expression a bit longer and we would need an eval to generate "reason" and "message" which would be clearer IMO.

Also consider using "search" rather then where to keep things consistent - as long as you don't need any eval functions, search should be sufficient.

File: querygen/querygen_test.go
Comment (ifireball) [inline querygen/querygen_test.go:23]: As far as I can tell we still need a test for `GenBuildPipelineRunStartedQuery` because we still use that function

**Review by ifireball — CHANGES_REQUESTED** [review]

File: querygen/querygen.go
Comment (amisstea) [reply querygen/querygen.go:76]: I made some suggestions on https://github.com/redhat-appstudio/segment-bridge/pull/36#discussion_r1304940201 related to these comments. PTAL and maybe we can incorporate them here.

**Comment by ShakedAviv50313** [comment]
i omitted the second `spath` and replaced `where` with `search`, the first spath is crucial as using mvexpand directly at responseObject.status.conditions{} without the spath returns:
``` 27 of 48 peers omitted, Field 'responseObject.status.conditions{}' does not exist in the data. ```

File: querygen/querygen_test.go
Comment (ShakedAviv50313) [reply querygen/querygen_test.go:23]: Done

**Comment by ShakedAviv50313** [comment]
@ifireball @amisstea WDYT about the changes?

File: querygen/querygen_test.go
Comment (amisstea) [reply querygen/querygen_test.go:23]: @ShakedAviv50313 we're now missing `TestGenBuildPipelineRunCreatedQuery`. Please ensure we have 3 tests for build PipelineRuns and that they call their respective query generation funcs.

File: querygen/querygen.go
Comment (amisstea) [reply querygen/querygen.go]: @ShakedAviv50313 have you attempted this?

cc @Kousalya1998 as it relates to #36

**Review by amisstea — CHANGES_REQUESTED** [review]

File: querygen/querygen.go
Comment (amisstea) [inline querygen/querygen.go:74]: We should be able to remove this `spath` command by using `responseObject.status.conditions{}.reason` instead of `reason`.

File: querygen/querygen.go
Comment (amisstea) [inline querygen/querygen.go:73]: This should filter by any defined completion time, not start time.

**Review by amisstea — CHANGES_REQUESTED** [review]
@ShakedAviv50313 please rebase your branch and resolve merge conflicts.

File: querygen/querygen.go
Comment (amisstea) [reply querygen/querygen.go:74]: To do this I think you'll need to "dedupe" the value like the FieldSet code does. E.g.
```
eval event="Build PipelineRun ".mvindex('responseObject.status.conditions{}.reason', 0)
```

File: querygen/querygen_test.go
Comment (ShakedAviv50313) [reply querygen/querygen_test.go:33]: Done, I rebased this branch with upstream/main

**Review by amisstea — APPROVED** [review]

**Comment by ShakedAviv50313** [comment]
@ifireball

File: querygen/querygen.go
Comment (ifireball) [inline querygen/querygen.go:106]: please break this long string

**Review by ifireball — APPROVED** [review]
Approving with the understanding that the code is going to be improved with work by @Kousalya1998

**Comment by ShakedAviv50313** [comment]
Last changes have been made, (breaking the long string).
I cannot merge this PR because I do not have write access to the repo

**Pull request merged**
